File: 프로그래머스/lv2/12949. 행렬의 곱셈/행렬의 곱셈.rb
def solution(arr1, arr2)
    ans = []
    0.upto(arr1.size-1) { |i|
        temp = []
        0.upto(arr2[0].size-1) { |j|
            var = 0
            0.upto(arr1[i].size-1) { |k| var += (arr1[i][k] * arr2[k][j]) }
            temp.push var
        }
        ans.push temp
    }
    ans
end

File: 프로그래머스/lv2/12949. 행렬의 곱셈/행렬의 곱셈.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<vector<int>> solution(vector<vector<int>> arr1, vector<vector<int>> arr2) {
    vector<vector<int>> answer;
    
    for(int i=0; i<arr1.size(); ++i) {
       vector<int> temp;
       for(int j=0; j<arr2[0].size(); ++j) {
           int x=0;
           for(int k=0; k<arr1[i].size(); ++k) {
               // cout << arr1[i][k] << ' ' << arr2[k][j] << endl;
               x += (arr1[i][k] * arr2[k][j]);
           }
           // cout << x << endl;
           temp.push_back(x);
       }
        answer.push_back(temp);
    }
    
    return answer;
}